Lib/ufo2ft/__init__.py

```diff
@@ -532,42 +532,83 @@ def compileVariableTTF(designSpaceDoc, **kwargs):
+    if len(fonts) != 1:
+        raise ValueError(
+            "Tried to build a DesignSpace version 5 with multiple variable "
+            "fonts using the old ufo2ft API `compileVariableTTF`. "
+            "Use the new API instead `compileVariableTTFs`"
+        )
+    return next(iter(fonts.values()))
 
-    excludeVariationTables = kwargs.pop("excludeVariationTables")
+
+compileVariableTTFs_args = {
+    **compileVariableTTF_args,
+    **dict(variableFontNames=None),
+}
+
+
+def compileVariableTTFs(designSpaceDoc: DesignSpaceDocument, **kwargs):
+    """Create FontTools TrueType variable fonts for each variable font defined
+    in the given DesignSpaceDocument, using their UFO sources
+    with interpolatable outlines, using fontTools.varLib.build.
+
+    *optimizeGvar*, if set to False, will not perform IUP optimization on the
+      generated 'gvar' table.
+
+    *excludeVariationTables* is a list of sfnt table tags (str) that is passed on
+      to fontTools.varLib.build, to skip building some variation tables.
+
+    *variableFontNames* is an optional list of names of variable fonts
+      to build. If not provided, all variable fonts listed in the given
+      designspace will by built.
+
+    The rest of the arguments works the same as in the other compile functions.
+
+    Returns a dictionary that maps each variable font filename to a new variable
+    TTFont object.
+
+    .. versionadded:: TODO (Jany) version of addition
+    """
+    kwargs = init_kwargs(kwargs, compileVariableTTFs_args)
     optimizeGvar = kwargs.pop("optimizeGvar")
+    excludeVariationTables = kwargs.pop("excludeVariationTables")
+    variableFontNames = kwargs.pop("variableFontNames")
 
-    # FIXME: Hack until we get a fontTools config module. Disable GPOS
-    # compaction while building masters because the compaction will be undone
-    # anyway by varLib merge and then done again on the VF
-    gpos_compact_value = os.environ.pop(GPOS_COMPACT_MODE_ENV_KEY, None)
-    try:
-        ttfDesignSpace = compileInterpolatableTTFsFromDS(
-            designSpaceDoc,
-            **{
-                **kwargs,
-                **dict(
-                    useProductionNames=False,  # will rename glyphs after varfont is built
-                    # No need to post-process intermediate fonts.
-                    postProcessorClass=None,
-                ),
-            },
-        )
-    finally:
-        if gpos_compact_value is not None:
-            os.environ[GPOS_COMPACT_MODE_ENV_KEY] = gpos_compact_value
+    # Pop inplace because we'll make a copy at this level so deeper functions
+    # don't need to worry
+    inplace = kwargs.pop("inplace")
+    if not inplace:
+        designSpaceDoc = designSpaceDoc.deepcopyExceptFonts()
+
+    vfNameToBaseUfo = _compileNeededSources(
+        kwargs, designSpaceDoc, variableFontNames, compileInterpolatableTTFsFromDS
+    )
+
+    if not vfNameToBaseUfo:
+        logger.warning("No variable fonts to build")
```

Comment:
i don't think it's necessary to warn, returning empty dict is ok. You may document in docstring that if no variable fonts are specifiedf in DS5, an empty dict is returned